packages/react/src/components/DataGrid/utils.tsx

```diff
@@ -52,15 +52,20 @@ export const useHeadlessColumns = <T extends Row>({
         id: HEADER_ID_SELECT,
         size: 34,
         header: () => null,
-        cell: ({ row }) => (
-          <Checkbox
-            checked={row.getIsSelected()}
-            disabled={!row.getCanSelect}
-            indeterminate={row.getIsSomeSelected()}
-            onChange={row.getToggleSelectedHandler()}
-            aria-label={t("components.datagrid.row_selection_aria")}
-          />
-        ),
+        cell: ({ row }) => {
+          if (!row.getCanSelect()) {
+            return null;
+          }
+          return (
+            <Checkbox
+              checked={row.getIsSelected()}
+              disabled={!row.getCanSelect}
```

Comment:
Your change highlight this logic is wrong
```suggestion
              disabled={!row.getCanSelect()}
```

Reply:
(ping) :)

Reply:
Woops! I did not our reply ...
Ok let's do that